libr/util/lib.c

```diff
@@ -303,19 +303,80 @@ R_API int r_lib_open_ptr (RLib *lib, const char *file, void *handler, RLibStruct
 
 	return ret;
 }
+#if __WINDOWS__ && !defined(__CYGWIN__)
+static char *utf16_to_utf8 (const wchar_t *wc) {
+	char *rutf8;
+	int csize;
+
+	if ((csize = WideCharToMultiByte (CP_UTF8, 0, wc, -1, NULL, 0, NULL, NULL))) {
+		if ((rutf8 = malloc (csize))) {
+			WideCharToMultiByte (CP_UTF8, 0, wc, -1, rutf8, csize, NULL, NULL);
+		}
+	}
+	return rutf8;
+}
+
+static wchar_t *utf8_to_utf16 (const char *cstring) {
+	wchar_t *rutf16;
+	int wcsize;
+
+	if ((wcsize = MultiByteToWideChar (CP_UTF8, 0, cstring, -1, NULL, 0))) {
+		if ((rutf16 = (wchar_t *) calloc (wcsize, sizeof (wchar_t)))) {
```

Comment:
This is still 'wrong', it can just be `if ((rutf16 = (wchar_t *) malloc (wcsize))) {